### PR TITLE
[Tabs] MDCTabBarViewController should delegate status bar to children

### PR DESCRIPTION
**Pull request by Icycle**
Because MDCTabBarViewController displays child view controllers that
control the top of the screen, it should delegate control of the
status bar to the currently selected child view controller, and ensure
that the status bar is updated when the selected child changes.

Fix for issue #2111 

#### Don't forget:
- [x] Identify the component the PR relates to in brackets in the title. ```[Buttons] Updated documentation```
- [x] Link to GitHub issues it solves. ```closes #1234```
- [ ] Sign the CLA bot. You can do this once the pull request is submitted.

[Contributing](./contributing/README.md#pull-requests) has more information and tips for a great
pull request.

### CODE REVIEW

**Comment by ianegordon** [comment]
@brianjmoore could you take a look at this?

File: components/Tabs/src/MDCTabBarViewController.m
Comment (brianjmoore) [inline components/Tabs/src/MDCTabBarViewController.m:254]: Maybe also `preferredStatusBarUpdateAnimation`?

**Review by brianjmoore — APPROVED** [review]
Thanks!

**Review by brianjmoore — APPROVED** [review]

File: components/Tabs/src/MDCTabBarViewController.m
Comment (Icycle) [reply components/Tabs/src/MDCTabBarViewController.m:254]: I think the default value is OK, but perhaps it could be overridden in a separate PR?

Comment (brianjmoore) [reply components/Tabs/src/MDCTabBarViewController.m:254]: Yeah I don't know if it's really necessary. I just see that it's done in some similar code I have open here, but that could be historical noise. 

Sounds good, let's just leave it to a separate PR if it seems to be necessary later.

**Comment by ianegordon** [comment]
I'll merge when Travis is happy.

**Comment by ianegordon** [comment]
Icycle is an internal dev.  Not sure why CLABot is unhappy.

**Pull request merged**
